discovery/client.go

```diff
@@ -132,6 +123,10 @@ func (r *defaultClientRegistrationManager) activate(ctx context.Context, service
 	if err := r.store.updatePresentationRefreshTime(serviceID, subjectID, parameters, &refreshVPAfter); err != nil {
 		return fmt.Errorf("unable to update Verifiable Presentation refresh time: %w", err)
 	}
+	// clear any previous presentationRefreshErrors here so it's triggered by both the refresh and API call
```

Comment:
The API calls `activate()`, the timer calls `refresh()` (which in turn calls `activate()`). `setPresentationRefreshError` is only called by `refresh()` if an error occurs (and never by `activate()`), this means;
- the API call will never cause the error to be set, even one of the DIDs fails and one of them succeeds
- the same applies for `refresh()`

Reply:
That's the idea. `activate()` via the API does not create a refresh record when it fails, so also not an error record. The error is passed back through the API.

So you only want to set an error (and thus being able to read it) when the refresh of a active registration fails.